grouper/fe/template_util.py

```diff
@@ -2,22 +2,28 @@
 
 from dateutil.relativedelta import relativedelta
 from jinja2 import Environment, PackageLoader
+from jinja2.runtime import Undefined
 from pytz import UTC
 from six import string_types
 
 from grouper.fe.settings import settings
 
 
-def _make_date_obj(date_obj):
+def _make_date_obj(input_date_obj):
     """Given either a datetime object, float date/time in UTC unix epoch, or
     string date/time in the form '%Y-%m-%d %H:%M:%S.%f' return a datetime
     object."""
-    if isinstance(date_obj, float):
-        date_obj = datetime.fromtimestamp(date_obj, UTC)
-
-    if isinstance(date_obj, string_types):
-        date_obj = datetime.strptime(date_obj, "%Y-%m-%d %H:%M:%S.%f")
-        date_obj = date_obj.replace(tzinfo=UTC)
+    if isinstance(input_date_obj, float):
+        date_obj = datetime.fromtimestamp(input_date_obj, UTC)
+    elif isinstance(input_date_obj, string_types):
+        try:
+            date_obj = datetime.strptime(input_date_obj, "%m/%d/%Y")
+        except ValueError:
+            date_obj = datetime.strptime(input_date_obj, "%Y-%m-%d %H:%M:%S.%f")
+    elif isinstance(input_date_obj, datetime):
+        date_obj = input_date_obj
+    else:
+        assert False, "{}>>>{}".format(input_date_obj, type(input_date_obj))
```

Comment:
I would add some text to this message that makes is more descriptive, maybe something like "date {} is of unsupported type {}"